quil-rs/src/program/scheduling/graph.rs

```diff
@@ -550,160 +474,18 @@ impl<'a> ScheduledProgram<'a> {
-                        instruction_index,
-                        custom_handler,
-                    )?;
-                }
-                Instruction::JumpWhen(JumpWhen {
-                    target: Target::Fixed(target),
-                    condition,
-                }) => {
-                    terminate_working_block(
-                        Some(BlockTerminator::Conditional {
-                            target,
-                            condition,
-                            jump_if_condition_true: true,
-                        }),
-                        &mut working_instructions,
-                        &mut blocks,
-                        &mut working_label,
-                        program,
-                        instruction_index,
-                        custom_handler,
-                    )?;
-                }
-                Instruction::JumpUnless(JumpUnless {
-                    target: Target::Fixed(target),
-                    condition,
-                }) => {
-                    terminate_working_block(
-                        Some(BlockTerminator::Conditional {
-                            target,
-                            condition,
-                            jump_if_condition_true: false,
-                        }),
-                        &mut working_instructions,
-                        &mut blocks,
-                        &mut working_label,
-                        program,
-                        instruction_index,
-                        custom_handler,
-                    )?;
-                }
-                Instruction::Halt => terminate_working_block(
-                    Some(BlockTerminator::Halt),
-                    &mut working_instructions,
-                    &mut blocks,
-                    &mut working_label,
-                    program,
-                    instruction_index,
-                    custom_handler,
-                )?,
-                Instruction::Gate(_)
-                | Instruction::Measurement(_)
-                | Instruction::Label(Label {
-                    target: Target::Placeholder(_),
-                })
-                | Instruction::Jump(_)
-                | Instruction::JumpWhen(_)
-                | Instruction::JumpUnless(_) => {
-                    return Err(ScheduleError {
-                        instruction_index,
-                        instruction: instruction.clone(),
-                        variant: ScheduleErrorVariant::UncalibratedInstruction,
-                    })
-                }
-            };
-        }
-
-        terminate_working_block(
-            None as Option<BlockTerminator>,
-            &mut working_instructions,
-            &mut blocks,
-            &mut working_label,
-            program,
-            None,
-            custom_handler,
-        )?;
-
-        Ok(ScheduledProgram { blocks })
+        let control_flow_graph = program.as_control_flow_graph();
```

Comment:
Note how much cleaner the implementation in `as_control_flow_graph` is than this one.

Also, that new implementation tolerates/elides more instructions than this previous one did. Some of those end up in errors within `ScheduledBasicBlock` because of lack of calibrations, but `DEFCAL` and friends are allowed now where they were once errors.